src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -665,6 +671,126 @@ void M1()
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[10]").WithLocation(6, 22)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_NoInitializer()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        /*<bind>*/int[10] x;/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationOperation (1 declarators) (OperationKind.VariableDeclaration, Type: null, IsInvalid) (Syntax: 'int[10] x')
+  Ignored Dimensions(1):
+      ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 10, IsInvalid) (Syntax: '10')
+  Declarators:
+      IVariableDeclaratorOperation (Symbol: System.Int32[] x) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'x')
+        Initializer: 
+          null
+  Initializer: 
+    null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,22): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         /*<bind>*/int[10] x;/*</bind>*/
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[10]").WithLocation(6, 22),
+                // file.cs(6,27): warning CS0168: The variable 'x' is declared but never used
+                //         /*<bind>*/int[10] x;/*</bind>*/
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "x").WithArguments("x").WithLocation(6, 27)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensionsWithInitializer_VerifyChildren()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        /*<bind>*/int[10] x = { 1 }/*</bind>*/;
+    }
+}
+";
+
+            var compilation = CreateEmptyCompilation(source);
+            (var operation, _) = GetOperationAndSyntaxForTest<VariableDeclarationSyntax>(compilation);
+            var declaration = (IVariableDeclarationOperation)operation;
+            Assert.Equal(2, declaration.Children.Count());
+            Assert.Equal(OperationKind.Literal, declaration.Children.First().Kind);
+            Assert.Equal(OperationKind.VariableDeclarator, declaration.Children.ElementAt(1).Kind);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_VerifyChildren()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        /*<bind>*/int[10] x;/*</bind>*/
```

Comment:
>10 [](start = 22, length = 2)

It would be good to have tests that try to get IOperation nodes for the sizes directly.